ksql-engine/src/test/java/io/confluent/ksql/codegen/SqlToJavaVisitorTest.java

```diff
@@ -180,4 +182,33 @@ public void shouldGenerateCorrectCodeForLikePatternWithoutWildcards() {
         .process(analysis.getWhereExpression());
     assertThat(javaExpression, equalTo("(TEST1_COL1).equals(\"foo\")"));
   }
+
+  @Test
+  public void shouldGenerateCorrectCodeForCaseStatement() {
+    // Given:
+    final Analysis analysis = analyzeQuery(
+        "SELECT CASE WHEN orderunits < 10 THEN 'small' WHEN orderunits < 100 THEN 'medium' ELSE 'large' END FROM orders;", metaStore);
+
+
+    // When:
+    final String javaExpression = new SqlToJavaVisitor(orderSchema, functionRegistry)
+        .process(analysis.getSelectExpressions().get(0));
+
+    // ThenL
+    assertThat(javaExpression, equalTo("(java.lang.String)SearchedCaseFunction.searchedCaseFunction(ImmutableList.of( SearchedCaseFunction.whenClause( new Supplier<Boolean>() { @Override public Boolean get() { return ((((Object)(ORDERS_ORDERUNITS)) == null || ((Object)(Integer.parseInt(\"10\"))) == null) ? false : (ORDERS_ORDERUNITS < Integer.parseInt(\"10\"))); }},  new Supplier<java.lang.String>() { @Override public java.lang.String get() { return \"small\"; }}),SearchedCaseFunction.whenClause( new Supplier<Boolean>() { @Override public Boolean get() { return ((((Object)(ORDERS_ORDERUNITS)) == null || ((Object)(Integer.parseInt(\"100\"))) == null) ? false : (ORDERS_ORDERUNITS < Integer.parseInt(\"100\"))); }},  new Supplier<java.lang.String>() { @Override public java.lang.String get() { return \"medium\"; }})), new Supplier<java.lang.String>() { @Override public java.lang.String get() { return \"large\"; }})"));
```

Comment:
The other expressions that result from SqlToJavaVisitor all appear to surround casts with an extra set of parentheses (for example, [here](https://github.com/confluentinc/ksql/blob/89d489ca3d817f979387ea2394cc69592b449649/ksql-engine/src/test/java/io/confluent/ksql/codegen/SqlToJavaVisitorTest.java#L94) or [here](https://github.com/confluentinc/ksql/blob/89d489ca3d817f979387ea2394cc69592b449649/ksql-engine/src/test/java/io/confluent/ksql/codegen/SqlToJavaVisitorTest.java#L128)). Is this extra set of parentheses necessary? (If so we should add them to `visitSearchedCaseExpression`.) I can’t think of an example in which they’d be required, but I’m also aware that translating ASTs to code can sometimes result in funkiness and that extra parentheses don’t hurt…

Reply:
Not necessary here but as you said it's better to be consistent and explicit. So I added the parentheses.